test/common/stats/thread_local_store_test.cc

```diff
@@ -2084,10 +2088,9 @@ TEST_P(HistogramParameterisedTest, ForEachSinkedHistogram) {
 TEST_P(HistogramParameterisedTest, UnsinkedHistogramsAreMerged) {
   StatNamePool pool(store_->symbolTable());
   store_->setSinkPredicates(std::make_unique<TestUtil::TestSinkPredicates>());
-  auto& sink_predicates =
-      static_cast<TestUtil::TestSinkPredicates&>(store_->sinkPredicates().ref());
+  auto& sink_predicates = sinkPredicates();
```

Comment:
actually I feel like this pattern has UB in a sense, because the store_ which owns the predicates, will be destroyed after the pool, and contains a StatNameHashSet that depends on the pool.

I think this problem was pre-existing, and benign because flat_hash_map's destructor must not need to call hash or equals functions on the keys. But I think it'd be better to put the pool into the TestSinkPredicates. WDYT?

Reply:
It was just easier to declare  pool_ in the test class before store_ than making the predicates own the pool and making that work with allocator_impl_test. Please let me know if that's acceptable.